src/components/date/__internal__/date-picker/date-picker.component.tsx

```diff
@@ -223,66 +191,108 @@ export const DatePicker = ({
     }
   };
 
-  const formatDay = (date: Date) =>
-    `${weekdaysShort[date.getDay()]} ${date.getDate()} ${
-      monthsShort[date.getMonth()]
-    } ${date.getFullYear()}`;
+  useEffect(() => {
+    if (selectedDays) {
+      setFocusedMonth(selectedDays);
+    }
+  }, [selectedDays]);
+
+  useEffect(() => {
+    if (!open && selectedDays) {
+      const fMonth = focusedMonth?.getMonth();
+      const sMonth = selectedDays?.getMonth();
+      if (fMonth !== sMonth) setFocusedMonth(selectedDays);
+    }
+  }, [focusedMonth, open, selectedDays]);
 
   if (!open) {
     return null;
   }
 
-  const localeUtils = { formatDay } as LocaleUtils;
-
   const handleTabGuardFocus = () => {
     ref.current?.querySelector("button")?.focus();
   };
 
   return (
-    <Popover
-      placement="bottom-start"
-      reference={inputElement}
-      middleware={popoverMiddleware}
-      disablePortal={disablePortal}
-    >
-      <StyledDayPicker
-        ref={ref}
-        onMouseDown={pickerMouseDown}
-        onKeyUp={handleKeyUp}
-        onKeyDown={handleOnKeyDown}
+    <>
+      <Popover
+        placement="bottom-start"
+        reference={inputElement}
+        middleware={popoverMiddleware}
+        disablePortal={disablePortal}
       >
-        <div
-          id={pickerTabGuardId}
-          // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
-          tabIndex={0}
-          onFocus={handleTabGuardFocus}
-        />
-        <DayPicker
-          month={selectedDays}
-          months={monthsLong}
-          firstDayOfWeek={weekStartsOn}
-          onDayClick={handleDayClick}
-          selectedDays={selectedDays}
-          weekdayElement={(weekdayElementProps) => {
-            const { className, weekday } = weekdayElementProps;
+        <StyledDayPicker
+          id="styled-day-picker"
+          ref={ref}
+          onMouseDown={pickerMouseDown}
+          onKeyUp={handleKeyUp}
+          onKeyDown={handleOnKeyDown}
+        >
+          <div
+            id={pickerTabGuardId}
+            // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
+            tabIndex={0}
+            onFocus={handleTabGuardFocus}
+          />
+          <DayPicker
+            required={false}
+            weekStartsOn={weekStartsOn}
+            onMonthChange={setFocusedMonth}
+            disabled={getDisabledDays(minDate, maxDate)}
+            locale={{
+              localize: {
+                ...defaultLocale.localize,
+                months: monthsLong,
+                weekdaysLong,
+                weekdaysShort,
+              },
+            }}
+            selected={focusedMonth}
+            month={focusedMonth}
+            onDayClick={(e) => {
```

Comment:
**suggestion**: I noticed when the date is cleared in the input box, the picker doesn't navigate to the current month as expected. What do you think about adding a default value for `month` to handle this case?:

```suggestion
            month={focusedMonth || new Date()}
```